torch/_inductor/virtualized.py

```diff
@@ -91,6 +93,102 @@ def inner(*args):
             setattr(cls, name, make_handler(format_string))
 
 
+class KernelBuilder:
```

Comment:
I'd suggest renaming this to KernelPrinter, since I believe it is only used in printing not in codegen.  (Note extract_read_writes just ignores the return value.)

In codegen we have a separate solution for this:
https://github.com/pytorch/pytorch/blob/ed41a7fb681ff51129763213042da0be4fbd1a56/torch/_inductor/codegen/common.py#L432-L488
and 
https://github.com/pytorch/pytorch/blob/ed41a7fb681ff51129763213042da0be4fbd1a56/torch/_inductor/codegen/common.py#L568-L612

Overall, I'd like to simplify this given it is just for printing code.  Perhaps we could reuse some of the code used in codgen?

A more minimal version might look something like:
```py
class AssignVars:
    def __init__(self, parent_handler):
       self.parent_handler = parent_handler
       self.output = IndentedBuffer()
       self.var_counter = itertools.count()

    def __getattr__(self, name):
        def inner(*args, **kwargs):
            line = getattr(self.parent_handler, name)(*args, **kwargs)
            if name == "indirect_indexing":
                return line
            # replace line with a new variable name
            varname = f"tmp{next(self.var_counter)}"
            self.output.writeline(f"{varname} = {line}")
            return varname
       
        return inner
```